### PR TITLE
Add ibmcloudmachineproviderspec

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderspec_types.go [hunk 1]
```diff
@@ -18,15 +18,36 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/sl"
```
Comment (xunpan) [inline pkg/apis/ibmcloud/v1alpha1/ibmcloudmachineproviderspec_types.go]: we should not import package which is not used

**Comment by gyliu513** [comment]
@jichenjc can you help rebase?

**Comment by jichenjc** [comment]
did the update

File: pkg/apis/apis.go [hunk 2]
```diff
@@ -22,11 +22,18 @@ package apis
 
 import (
 	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api-provider-openstack/pkg/apis/openstackproviderconfig/v1alpha1"
```
Comment (gyliu513) [inline pkg/apis/apis.go]: Here should be ibmcloud, I have updated.

File: pkg/apis/ibmcloud/v1alpha1/zz_generated.deepcopy.go [hunk 3]
```diff
@@ -214,7 +214,7 @@ func (in *IbmcloudMachineProviderSpec) DeepCopyInto(out *IbmcloudMachineProvider
 	*out = *in
 	out.TypeMeta = in.TypeMeta
 	in.ObjectMeta.DeepCopyInto(&out.ObjectMeta)
-	out.Spec = in.Spec
+	in.Spec.DeepCopyInto(&out.Spec)
```
Comment (gyliu513) [inline pkg/apis/ibmcloud/v1alpha1/zz_generated.deepcopy.go]: Does the update of the file auto-generated?

Besides, I have add some new files to check gofmt, golint at `hack/xxx`, we can run those scripts manually for now, but I will put them to makefile later.

**Comment by jichenjc** [comment]
I did run test locally and this vet test passed before I submit patch
I will double check again

**Pull request merged**